plugin/evm/syncervm_client.go

```diff
@@ -108,6 +123,71 @@ type Syncer interface {
 	Done() <-chan error
 }
 
+// AsyncReceive returns true if the client is ready to receive a message from the engine
+// Should return true if syncing and useUpstream is true, i.e. currently dynamicaling syncing
+func (client *stateSyncerClient) AsyncReceive() bool {
+	return client.useUpstream && client.syncing.Get() && client.dl != nil && !client.dl.execQueue.Closed()
+}
+
+func (client *stateSyncerClient) QueueVerifyBlock(b *Block) error {
+	if !client.AsyncReceive() {
+		return fmt.Errorf("cannot queue block when not using upstream syncing")
+	}
+
+	verify := func(a *Block) error {
+		return a.VerifyDuringSync(context.TODO()) // is this right?
```

Comment:
this is not gonna work since during sync the state is incomplete. let's chat about it.